### ISSUE TITLE
Attempting to rename a samba share deletes all contents in the folder

### ISSUE THREAD

**Issue by gitinthebus** [nextcloud/server#1839]
### Steps to reproduce
1. Using the external storage option under admin menu point to a samba share with username and password authentication
2. Go back to "Files" Add a few files in the shared samba folder if none exists
3. While still browsing "all files" try to rename the samba share using the "..." button
### Expected behaviour

It should rename the folder/share
### Actual behaviour

It fails to rename it and **deletes all contents**
### Server configuration

**Operating system**:
Linux
**Web server:**
Apache
**Database:**
SQLite
**PHP version:**
5.6.26
**Nextcloud version:** (see Nextcloud admin page)
10.0 Stable
**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh 
**Where did you install Nextcloud from:**
Docker

**Comment by gitinthebus** [comment]
Are you considering to fix this bug? 
It seems critical to me since it's causing data loss.

**Comment by patrick1990** [comment]
I reproduced the issue with a local external storage.
If a user renames the external folder via the files app, nextcloud moves the content of the external storage to the users file directory (<nextcloud-data-dir>/<username>/files). This leads to errors with the external storage app (because the external folder is gone).
Furthermore if an external directory is made available to multiple users, every user can revoke access to the external folder for the other users via this issue. For instance:

1. Admin adds external storage "Project-Files" to User1 and User2.
2. User1 renames "Project-Files" via the files app to "My-Project".
3. User2 has no access to "Project-Files" anymore and gets an error from the external storage app, becaus the directory of "Project-Files" was moved to the user directory of User1.

Maybe renaming of external folders should not be allowed via the files app?
